File: LeetCode/1.js
// my answer
let twoSum = function(nums, target) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = i+1; j < nums.length; j++) {
      if ((nums[i] + nums[j]) === target) {
        return [i, j]
      }
    }
  }
  return false
};

// opt answer
// HashMap
let twoSum = function(nums, target) {
  let map = new Map();
  for (let i = 0; i < nums.length; i++) {
    let gapValue = target - nums[i];
    if (map.get(gapValue) !== undefined) {
        return [map.get(gapValue), i];
    } else {
        map.set(nums[i], i);
    }
  }
  return false
};

File: LeetCode/Array/1.go
package main

import "fmt"

func main() {
	nums := []int{2, 7, 11, 15}
	target := 18
	twoSum(nums, target)
}

func twoSum(nums []int, target int) []int {
	tempMap := map[int]int{}
	for i, v := range nums {
		g := target - v

		if j, ok := tempMap[g]; ok {
			return []int{j, i}
		}
		tempMap[v] = i
	}
	return []int{}
}
